crates/cheatnet/src/runtime_extensions/forge_runtime_extension/mod.rs

```diff
@@ -686,10 +687,22 @@ fn handle_deploy_result(
 fn serialize_call_trace(call_trace: &CallTrace, output: &mut Vec<Felt252>) {
     serialize_call_entry_point(&call_trace.entry_point, output);
 
-    output.push(Felt252::from(call_trace.nested_calls.len()));
+    let visible_calls: Vec<_> = call_trace
+        .nested_calls
+        .iter()
+        .filter_map(|call_node| {
+            if let CallTraceNode::EntryPointCall(call) = call_node {
+                Some(call)
+            } else {
+                None
+            }
+        })
+        .collect();
```

Comment:
I don't understand what `visible` means in this context.

Reply:
Visible to the user, we don't want to put nodes that has no call (deploy without constructor) to cairo

Reply:
Disagree, it shouldn't appear in cairo at all, I would have to expose this type in snforge_std then

Reply:
and why this is a problem? this way you hide this detail deep in the internals

Reply:
I can create an issue for that, but it's out of scope of this PR, I don't want to waste time on this

Reply:
https://github.com/foundry-rs/starknet-foundry/issues/2000